File: src/2025-02-22-00-51-50--a-deepseek-r1-test-220-10-1--2-1-huoshan/javascript/exercises/practice/bottle-song/bottle-song.js
//
// This is only a SKELETON file for the 'Bottle Song' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const recite = (initialBottlesCount, takeDownCount) => {
  const numberWords = ['', 'One', 'Two', 'Three', 'Four', 'Five', 'Six', 'Seven', 'Eight', 'Nine', 'Ten'];
  const verses = [];
  
  for (let i = 0; i < takeDownCount; i++) {
    const current = initialBottlesCount - i;
    const currentWord = numberWords[current];
    const currentBottle = current === 1 ? 'bottle' : 'bottles';
    
    const nextNumber = current - 1;
    const nextWord = nextNumber === 0 ? 'no' : numberWords[nextNumber].toLowerCase();
    const nextBottle = nextNumber === 1 ? 'bottle' : 'bottles';
    
    verses.push(
      `${currentWord} green ${currentBottle} hanging on the wall,`,
      `${currentWord} green ${currentBottle} hanging on the wall,`,
      `And if one green bottle should accidentally fall,`,
      `There'll be ${nextWord} green ${nextBottle} hanging on the wall.`
    );
    
    // Add empty string between verses except after last one
    if (i < takeDownCount - 1) {
      verses.push('');
    }
  }
  
  return verses;
};

File: src/2025-02-22-00-51-50--a-deepseek-r1-test-220-10-1--2-1-huoshan/python/exercises/practice/bottle-song/bottle_song.py
def recite(start, take=1):
    verses = []
    number_words = ["Two", "Three", "Four", "Five", "Six", "Seven", "Eight", "Nine", "Ten"]
    
    for current in range(start, start - take, -1):
        if current < 1:
            break
        
        # Generate current bottle phrase
        if current == 1:
            current_phrase = "One green bottle"
        elif 2 <= current <= 10:
            current_phrase = f"{number_words[current - 2]} green bottles"
        else:
            current_phrase = f"{current} green bottles"
        
        # Generate next bottle phrase
        next_n = current - 1
        if next_n == 0:
            next_phrase = "no green bottles"
        elif next_n == 1:
            next_phrase = "one green bottle"
        elif 2 <= next_n <= 10:
            next_phrase = f"{number_words[next_n - 2].lower()} green bottles"
        else:
            next_phrase = f"{next_n} green bottles"
        
        verse = [
            f"{current_phrase} hanging on the wall,",
            f"{current_phrase} hanging on the wall,",
            "And if one green bottle should accidentally fall,",
            f"There'll be {next_phrase} hanging on the wall."
        ]
        verses.append(verse)
    
    # Flatten verses with empty line separators
    lines = []
    for i, verse in enumerate(verses):
        lines.extend(verse)
        if i != len(verses) - 1:
            lines.append("")
    
    return lines
